diff --git a/src/components/bluepad32/CMakeLists.txt b/src/components/bluepad32/CMakeLists.txt
index 175d0e0..73f5648 100644
--- a/src/components/bluepad32/CMakeLists.txt
+++ b/src/components/bluepad32/CMakeLists.txt
@@ -169,9 +169,12 @@ if(IDF_TARGET)
     # ESP-IDF
     # Nothing
 elseif(PICO_SDK_VERSION_STRING)
+    if(NOT DEFINED PICO_CYW43_ARCH_LIBRARY)
+        set(PICO_CYW43_ARCH_LIBRARY pico_cyw43_arch_none CACHE INTERNAL "CYW43 arch target for Bluepad32")
+    endif()
     target_link_libraries(bluepad32
             pico_stdlib
-            pico_cyw43_arch_none
+            ${PICO_CYW43_ARCH_LIBRARY}
             pico_btstack_ble
             pico_btstack_classic
             pico_btstack_cyw43
